src/main/java/Duke.java

```diff
@@ -1,10 +1,178 @@
+            "    5. list = lists out all current tasks with their taskIndex\n" +
+            "    6. echo = turn into a huge cave and echo your inputs back to you!!\n" +
+            "    7. quit (only in echo mode) = turn back to normal and stop echoing\n" +
+            "    8. genshin = begin the genshin helper\n" +
+            "    9. bye = shuts me down... ;-;\n";
+
+
+    private static final Scanner sc = new Scanner(System.in);
+    
+
+
+    //Prints greeting message
+    public static void greeting() {
+        System.out.println(LINES + GREETING + LINES);
+    }
+
+    //Begins echo function
+    public static void echo() {
+        System.out.println(LINES + ECHO_MESSAGE + LINES);
+        boolean echoState = true;
+        while (echoState) {
+            String s = sc.nextLine();
+            if (s.equalsIgnoreCase("quit")) {
+                System.out.println(LINES + QUIT_MESSAGE + LINES);
+                echoState = false;
+            } else {
+                String output = LINES + " " + s + "\n" + LINES;
+                System.out.println(output);
+            }
+        }
+    }
+
+    //list function
+    public static void list(Task[] tasks) {
+        System.out.print(LINES);
+        for (int i = 0; tasks[i] != null; i++) {
+            int j = i + 1;
+            System.out.println(j + "." + tasks[i]);
+        }
+        System.out.print(LINES);
+    }
+
+    //Marks a specified task input as done
+    public static void markDone(String input, Task[] tasks, int taskCount) {
+        //isolate 'x' from 'done x', where x is a number
+        int index = Integer.parseInt(input.substring(5));
+        if (index > taskCount) {
+            System.out.println(LINES + NO_TASK_MESSAGE + LINES);
+        }
+        else {
+            tasks[index - 1].markAsDone();
+            System.out.print(LINES + TASK_COMPLETE_MESSAGE + tasks[index-1] + "\n" + LINES);
+        }
+    }
+    //Adds task to tasks
+    public static int addTask(Task[] tasks, String input, int taskCount) {
+        if (input.toUpperCase().startsWith("TODO ")) {
+            tasks[taskCount] = new Todo(input.substring(5));
+        }
+        else if (input.toUpperCase().startsWith("DEADLINE ")) {
+            int slashIndex = input.indexOf("/");
+            String task = input.substring(9, (slashIndex-1));
+            String dueDate = input.substring(slashIndex+4);
+            tasks[taskCount] = new Deadline(task, dueDate);
+        }
+        else {
+            int slashIndex = input.indexOf("/");
+            String task = input.substring(6, (slashIndex-1));
+            String timeRange = input.substring(slashIndex+4);
+            tasks[taskCount] = new Event(task, timeRange);
+        }
+        System.out.println(LINES + ADD_MESSAGE + tasks[taskCount]);
+        taskCount += 1;
+        System.out.println(" Now you have " + taskCount + " tasks in the list.\n" + LINES);
+        return taskCount;
```

Comment:
Good use of if and else ifs to avoid deep nesting, but perhaps you could extract this method down even further as it is a little lengthy.